sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -562,6 +568,35 @@ private bool TryLoadConstructorArguments(AttributeData attributeData, IDictionar
                 return true;
             }
 
+            private bool TryOverrideBindingName(INamedTypeSymbol? attributeClass, string argumentName, out string? newName)
```

Comment:
You can change this signature to a ref to remove the if statement usage of it:
```suggestion
            private void OverrideBindingName(INamedTypeSymbol? attributeClass, ref string argumentName)
```

with `ref` you can choose to update that value or not. So the usage will be:

``` CSharp
string argumentName = attribMethodSymbol.Parameters[i].Name;
OverrideBindingName(attributeData.AttributeClass, argumentName); // either argumentName will remain unchanged OR be updated to the overridden name at the end of this.
```